tasks/collect_api_events.rb

```diff
@@ -0,0 +1,60 @@
+require 'common_events_library'
+require 'benchmark'
+require 'json'
+require 'net/https'
+require 'time'
+
+# params = JSON.parse(STDIN.read)
+
+params = {}
+
+pe_console = params['pe_console'] || 'runaway-causing.delivery.puppetlabs.net'
+pe_username = params['pe_username'] || 'admin'
+pe_password = params['pe_password'] || 'pie'
+splunk_server = params['splunk_server'] || 'localhost'
+splunk_token  = params['splunk_token'] || 'abcd1234'
+splunk_port = params['splunk_port'] || '8088'
+
+def transform(body, pe_console, source_type, splunk_client, splunk_token)
+  events_json = ''
+  body.each do |pe_event|
+    timestamp = Time.new
+    event = {
+      'host' => pe_console,
+      'time' => timestamp.to_i,
+      'sourcetype' => source_type,
+      'event' => pe_event,
+    }
+    splunk_client.post_request('/services/collector', event, {'Authorization': "Splunk #{splunk_token}"})
```

Comment:
Can we move the post out of the transform? I kind of think that a transform function that actually does the post is maybe a little unexpected.

Reply:
The next PR will move the post out of transform when I complete the batching. You are corret that the post should not remain in the transform.